src/version_edit.h

```diff
@@ -33,6 +35,7 @@ class VersionEdit {
 
  private:
   friend class VersionSet;
+  friend class EditCollector;
 
   bool has_next_file_number_{false};
   uint64_t next_file_number_{0};
```

Comment:
I think the logic is cleaner if `EditCollector` also handle `next_file_number_`.

Reply:
What do you mean? `EditCollector` checks whether `next_file_number_` is monotonic increasing?

Reply:
Oh, I got it.